File: Top Interview 150/Python/Remove_Element.py
class Solution:
    def removeElement(self, nums: list[int], val: int) -> int:
        while(val in nums):
            nums.remove(val)
        return len(nums)

print(Solution().removeElement([3,2,2,3], 3)) # 2
print(Solution().removeElement([0,1,2,2,3,0,4,2], 2)) # 5
print(Solution().removeElement([3,3], 3)) # 0

File: Top Interview 150/C++/Remove_Element.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        for(short i = 0;i < nums.size();++i){
            if (nums[i] == val)nums.erase(i + nums.begin()),--i;
        }
        return nums.size();
    }
};

int main(){
    Solution s;
    vector <int> nums = {3,2,2,3};
    int val = 3;
    cout << s.removeElement(nums,val) << endl; // 2
    return 0;
}